app/backend/src/couchers/servicers/account.py

```diff
@@ -150,3 +157,61 @@ def MarkContributorFormFilled(self, request, context):
             user = session.query(User).filter(User.id == context.user_id).one()
             user.filled_contributor_form = request.filled_contributor_form
         return empty_pb2.Empty()
+
+    def ChangePhone(self, request, context):
+        phone = request.phone
+        # early quick validation
+        if phone and not is_e164_format(phone):
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.INVALID_PHONE)
+
+        with session_scope() as session:
+            user = session.query(User).filter(User.id == context.user_id).one()
+            if not phone:
+                user.phone = None
+                user.phone_verification_verified = None
+                user.phone_verification_token = None
+                return empty_pb2.Empty()
+
+            if not is_known_operator(phone):
+                context.abort(grpc.StatusCode.UNIMPLEMENTED, errors.UNRECOGNIZED_PHONE_NUMBER)
+
+            if user.phone_verification_sent and now() - user.phone_verification_sent < timedelta(days=180):
+                context.abort(grpc.StatusCode.RESOURCE_EXHAUSTED, errors.REVERIFICATION_TOO_EARLY)
+
+            token = sms.generate_random_code()
+            result = sms.send_sms(phone, sms.format_message(token))
+
+            if result == "success":
+                user.phone = phone
+                user.phone_verification_sent = now()
+                user.phone_verification_token = token
+                user.phone_verification_attempts = 0
+                return empty_pb2.Empty()
+
+        context.abort(grpc.StatusCode.UNIMPLEMENTED, result)
+
+    def VerifyPhone(self, request, context):
```

Comment:
The could should also expire, say after 24 hours.

Reply:
?SYNTAX ERROR

Reply:
Ahahahah, sorry.

I mean: the verification code should expire if the user doesn't `VerifyPhone` within some reasonable amount of time.

(That way we don't end up with this... https://bugzilla.mozilla.org/show_bug.cgi?id=1630870 [summary: GlobalSign issued a certificate with completely obsolete and non-(today's-)standards compliant cert because someone clicked a 9 year old email])

Reply:
(I meant this one: https://bugzilla.mozilla.org/show_bug.cgi?id=1690807 wrong cert bug lol)